editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/internal/ProjectTemplatesActivator.java

```diff
@@ -46,41 +63,65 @@ public ProjectTemplatesActivator() {
 	public static ProjectTemplatesActivator getDefault() {
 		return instance;
 	}
-	
+
 	public static BundleContext getBundleContext() {
 		return instance.getBundle().getBundleContext();
 	}
 
 	/*
 	 * (non-Javadoc)
-	 * @see org.eclipse.ui.plugin.AbstractUIPlugin#start(org.osgi.framework.BundleContext)
+	 * 
+	 * @see org.eclipse.ui.plugin.AbstractUIPlugin#start(org.osgi.framework.
+	 * BundleContext)
 	 */
 	@Override
-    public void start(BundleContext context) throws Exception {
-        super.start(context);
-        registerDebugOptionsListener(PLUGIN_ID, new Trace(this), context);
+	public void start(BundleContext context) throws Exception {
+		super.start(context);
+		registerDebugOptionsListener(PLUGIN_ID, new Trace(this), context);
+		registerWorkspaceListener();
 	}
-    
-    @Override
-    protected BaseUISharedImages createSharedImages() {
-    	return new ProjectTemplatesSharedImages(getBundle());
-    }
 	
+	/* (non-Javadoc)
+	 * @see org.eclipse.ui.plugin.AbstractUIPlugin#stop(org.osgi.framework.BundleContext)
+	 */
+	@Override
+	public void stop(BundleContext context) throws Exception {
+		unregisterWorkspaceListener();
+		super.stop(context);
+	}
+
+	@Override
+	protected BaseUISharedImages createSharedImages() {
+		return new ProjectTemplatesSharedImages(getBundle());
+	}
+
+	private void registerWorkspaceListener() {
+		listener = new POMChangedReporter();
+		ResourcesPlugin.getWorkspace().addResourceChangeListener(listener,
+				IResourceChangeEvent.POST_BUILD | IResourceChangeEvent.POST_CHANGE);
+	}
+
+	private void unregisterWorkspaceListener() {
+		ResourcesPlugin.getWorkspace().removeResourceChangeListener(listener);
+		listener = null;
+	}
+
 	/**
 	 * Gets message from plugin.properties
+	 * 
 	 * @param key
 	 * @return
 	 */
-	public static String getMessage(String key)	{
+	public static String getMessage(String key) {
 		return Platform.getResourceString(instance.getBundle(), key);
 	}
 
 	/**
-	 * Get the IPluginLog for this plugin. This method 
-	 * helps to make logging easier, for example:
+	 * Get the IPluginLog for this plugin. This method helps to make logging easier,
+	 * for example:
+	 * 
+	 * FoundationCorePlugin.pluginLog().logError(etc)
```

Comment:
it was already there before but the example is referencing FoundationCorePlugin.pluginLog().logError(etc) instead of ProjectTemplatesActivator.pluginLog().logError(etc)